商品出品機能

shivonnu opened the pull request:
# WHAT
画像つきで商品の情報が保存できるように商品出品機能を実装します。
# WHY
フリマアプリの商品出品機能部分を実装する為です。
↓テストコード実施後のターミナル画面の画像です。
https://gyazo.com/36fd25567442d40c574a73eb5edcc664
以下gyazo動画リンクです。
↓出品して保存された所です。
https://gyazo.com/f6102f8e3efa84daa5159a2067253941
↓名前だけで出品してエラー表示をしてる所です。
https://gyazo.com/f758d8c989a9c4f8d1c1398c277f2be9

shivonnu committed:
gem rails_12factorの追加

shivonnu committed:
first commit

Itemテーブルを作成、userとアソシエーションをつなぎました。
userテーブルだけの時はuserのモデルとコントローラーでindex出してたのでitemテーブルに繋ぎ直してuserのビューは削除しました。

shivonnu committed:
Active Storageの導入

画像を表示させる為にActive Storageを導入しました。

shivonnu committed:
アクティブハッシュ組み込み、ルーティング決め

アクティブハッシュを使いカテゴリなどの選択肢表示をできるようにしました。アクティブストレージ導入、ルートコントローラー編集して画像とデータがDBに保存できるようにしました。

shivonnu committed:
モデル単体テストコードと、画像が無いと出品できないvalidation設定

各モデル単体テストコード入れて確認しました。Active Storageのテストコードは実装されてないとの事でしたので、手で確かめて画像がないと投稿できないことを確認しました。

shivonnu committed:
値段を表示させる為にjavascriptファイル設置、yarnとwebpacker再インストール,エラー表示

税抜きの値段を表示させる為にwebpecker周りのファイルを設置。requie""が読み込めなくてメソッドが見つからないエラーが続いたので、yarnを再インストールしてwebpeckerを再インストールしました。エラー文を表示させました。

shivonnu committed:
モデル単体テストコード用の画像の追加

javascriptのくだりでテスト用の画像を消してしまっていたので、戻しました。テストコード実施、ルボコップも使用しました。

i-date commented on app/controllers/items_controller.rb:
```diff
@@ -0,0 +1,35 @@
+class ItemsController < ApplicationController
+  def index
+    @item = Item.all
```
indexアクションについて、このプルリクエストでレビューしてしまうと、今後のプルリクエストで拝見したいファイルが差分として上がってこない可能性があります。
次回以降の商品一覧機能実装に対応するプルリクエストで拝見させていただきたいので、一旦こちらはコメントアウトしておくといいでしょう。

i-date commented on config/routes.rb:
```diff
@@ -1,6 +1,7 @@
-  root to: "users#index"
+  resources :items
+  root to: "items#index"
+  post "items/new", to: "items#create"
```
3行目の`resources :items`にて作成されるルーティングがあれば、5行目のルーティングは不要ではないでしょうか？

i-date commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,65 @@
+    before do
+      @item = FactoryBot.build(:item)
+    end
+    describe '商品出品機能' do
```
「すべての情報が正しいフォーマットで入力されていれば出品登録できる」というテストも追加しましょう。

i-date reviewed (commented):
レビュー依頼ありがとうございます！
コメントいたしましたので、ご確認をお願いいたします。

i-date commented:
また、実装いただいた単体テスト結果を確認するため、上記修正後に、お手数ですがプルリクにターミナルで実行したテスト結果のキャプチャ（のURLなど）を載せていただけないでしょうか？

shivonnu committed:
indexルーティングコメントアウト、post...の削除,テストコード追加

indexルーティングを次の工程で反映することが望ましいのでコメントアウトしました。post "items/new", to: "items#create"不要の為削除しました。全てOKなら登録できるテストを追加しました。

shivonnu commented:
コメントありがとうございます！
修正、また画像URLをプルリクエストに上げましたので再度コードレビュー依頼させて頂きます！

syuji-yamamoto commented on app/views/items/index.html.erb:
```diff
@@ -182,7 +182,7 @@
 <div class='purchase-btn'>
   <span class='purchase-btn-text'>出品する</span>
-  <a href="#">
+  <a href="/items/new">
```
aタグはlink_toに書き換えましょう！
またパスはprefixを使用しましょう！

syuji-yamamoto commented on app/views/items/new.html.erb:
```diff
@@ -1,14 +1,18 @@
     <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
-    <%# render 'shared/error_messages', model: f.object %>
+    <%= render 'shared/error_messages', model: f.object  %>
     <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
実装が完了したら、可読性向上のためコメントアウトは削除しておきましょう！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
細かい点ですが、コメントしましたので、ご確認ください！

shivonnu committed:
aタグのリンクをlink_to,prefixパス表記に変更、コメントアウト削除

商品出品ページへのaタグのリンクをlink_to,prefixパス表記に変更、コメントアウト作業終了部分削除しました。コメントアウト削除中新規登録のエラー表記処理してないのに気づけたので表示する設定をしました。

shivonnu commented:
コメントありがとうございます！
修正しましたので再度コードレビュー依頼させて頂きます！

syuji-yamamoto commented:
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
指摘箇所以外のコメントアウトも削除していただいたのですね！
素晴らしいです！これからも可読性を意識してコードを書いていきましょう！！

LGTMとさせていただきます！実装お疲れ様でした！
引き続き、実装頑張っていきましょう！

shivonnu commented:
ありがとうございます！
引き続き頑張ります！
